arangod/Aql/Executor/JoinExecutor.cpp

```diff
@@ -76,7 +77,11 @@ JoinExecutor::JoinExecutor(Fetcher& fetcher, Infos& infos)
     : _fetcher(fetcher),
       _infos(infos),
       _trx{_infos.query->newTrxContext()},
-      _resourceMonitor(_infos.query->resourceMonitor()) {
+      _resourceMonitor(_infos.query->resourceMonitor()),
+      _projectionsBuilder(std::make_shared<velocypack::SupervisedBuffer>(
```

Comment:
Can't the supervised buffer be built on the stack as _projectionsBuilder is a class member? Will its buffer get stolen?

Reply:
I can be done as a class member but this version is also fine, the old constructor would have created a SharedPtr also so I am good here.